README.md

```diff
@@ -1,80 +1,173 @@
 # Knative Kafka
 
-This project is a Knative Eventing implementation of a Kafka backed channel which provides advanced functionality and
-production grade qualities.
-
-**NOTE** - The latest code on the /master branch is compatible with Knative-Eventing v0.12.0.  We are currently working
-to catch up with Knative-Eventing /master.  The plan is to release v0.12.0 support soon, and then move the implementation
-on /master to align with the most recent from knative-eventing.
-
-The following components comprise the total **Knative-Kafka** solution...
-
-- [knative-kafka-common](./components/common/README.md) - Common code used by the other Knative-Kafka components.
-
-- [knative-kafka-channel](./components/channel/README.md) - The event receiver of the Channel to which inbound
-messages are sent.  An http server which accepts messages that conform to the CloudEvent specification and then writes
-those messages to the corresponding Kafka Topic. This is the "Producer" from the Kafka perspective.
+This project is a Knative Eventing implementation of a Kafka backed channel 
+which provides advanced functionality and production grade qualities beyond
+what the [eventing-contrib/kafka](https://github.com/knative/eventing-contrib/tree/master/kafka) 
+implementation offers.
+
+
+## Rationale
+
+The Knative eventing-contrib repository already contains a Kafka implementation, 
+so why invest the time in building another one?  At the time this project was 
+begun, and still today, the reference Kafka implementation does not provide the
+scaling characteristics required by a large and varied use case with many 
+different Topics and Consumers.  That implementation is based on a single 
+choke point that could easily allow one Topic's traffic to impact the 
+throughput of another Topic.  It is also not horizontally scalable as it only 
+supports a single instance of the dispatcher/consumer.  Further, no ordering 
+guarantees on the consumption side are provided which is required in certain 
+use cases.  The reference implementation was also based on the 
+[Sarama Go Client](https://github.com/Shopify/sarama) which has limitations 
+compared to the [Confluent Go Client](https://github.com/confluentinc/confluent-kafka-go), 
+and the [librdkafka C library](https://github.com/edenhill/librdkafka) on which 
+it is based.  Finally, the ability to expose Kafka configuration was very 
+limited, and we needed the ability to customize certain aspects of the Kafka 
+Topics / Producers / Consumers.
+
```

Comment:
Should probably add sentence or two about supporting a variety of kafka providers

Reply:
👍